deps/rabbit/src/rabbit_disk_monitor.erl

```diff
@@ -264,11 +250,37 @@ parse_free_unix(Str) ->
         _          -> exit({unparseable, Str})
     end.
 
-parse_free_win32(CommandResult) ->
-    LastLine = lists:last(string:tokens(CommandResult, "\r\n")),
-    {match, [Free]} = re:run(lists:reverse(LastLine), "(\\d+)",
-                             [{capture, all_but_first, list}]),
-    list_to_integer(lists:reverse(Free)).
+win32_get_disk_free_fsutil(Dir) ->
+    % Dir:
+    % "c:/Users/username/AppData/Roaming/RabbitMQ/db/rabbit2@username-z01-mnesia"
+    Drive = string:slice(Dir, 0, 2),
+
+    % Drive: c:
+    FsutilCmd = "fsutil volume diskfree " ++ Drive,
+
+    % C:\windows\system32>fsutil volume diskfree c:
+    % Total free bytes        :   812,733,878,272 (756.9 GB)
+    % Total bytes             : 1,013,310,287,872 (943.7 GB)
+    % Total quota free bytes  :   812,733,878,272 (756.9 GB)
+    FsutilResult = run_cmd(FsutilCmd),
+    case string:slice(FsutilResult, 0, 5) of
+        "Error" ->
+            error;
+        "Total" ->
+            FirstLine = hd(string:tokens(FsutilResult, "\r\n")),
+            {match, [FreeStr]} = re:run(FirstLine, "(\\d+,?)+", [{capture, first, list}]),
+            {ok, list_to_integer(lists:flatten(string:tokens(FreeStr, ",")))}
```

Comment:
A fun way to do this would be:

```
list_to_integer(FreeStr -- ",,,,,,,,,,,,")
```

:-)

Reply:
That's pretty darn clever.